smk/QC_2.smk

```diff
@@ -87,85 +86,55 @@ for i in ilmn_samples:
 # Host genome filtering
 ##############################################
 
-host_genome_path=''
-if config['PATH_host_genome'] is None:
-    print('ERROR in ', config_path, ': PATH_host_genome variable is empty. Please, complete ', config_path)
-elif path.exists(config['PATH_host_genome']) is False:
-    print('ERROR in ', config_path, ': PATH_host_genome variable path does not exit. Please, correct ', config_path)
-else:
-    host_genome_path=config["PATH_host_genome"]
-
-host_genome_name=''
-if config['NAME_host_genome'] is None:
-    print('ERROR in ', config_path, ': NAME_host_genome variable is empty. Please, complete ', config_path)
-elif path.exists(host_genome_path+'/BWA_index/'+config['NAME_host_genome']+'.pac') is True:
-    host_genome_name=config["NAME_host_genome"]
-    print('Host genome db {dir}/BWA_index/{db} will be used'.format(db=host_genome_name, dir=host_genome_path))
-elif path.exists(host_genome_path+'/'+config['NAME_host_genome']) is True:
-    host_genome_name=config["NAME_host_genome"]
-    print('Host genome sequence {dir}/{db} will be used to create BWA db'.format(db=host_genome_name, dir=host_genome_path))
+host_genome_path = validate_required_key(config, 'PATH_host_genome')
+host_genome_name = validate_required_key(config, 'NAME_host_genome')
+
+if path.exists(f"{host_genome_path}/BWA_index/{host_genome_name}.pac") is True:
+    print(f"Host genome db {host_genome_path}/BWA_index/{host_genome_name} will be used")
+elif path.exists(f"{host_genome_path}/{host_genome_name}") is True:
+    print(f"Host genome sequence {host_genome_path}/{host_genome_name} will be used to create BWA db")
 else:
-    print('ERROR in {}: NAME_host_genome={} does not exist as fasta or BWA db in PATH_host_genome={}. Please, correct!'.format(config_path, config['NAME_host_genome'], config['PATH_host_genome']))
+    raise Exception(f"NAME_host_genome={host_genome_name} does not exist as fasta or BWA db in PATH_host_genome={host_genome_path}. Please fix {config_path}")
 
 ##############################################
 # Minimum read-length trimming
 ##############################################
 
-read_min_len = 50
-if 'READ_minlen' in config and config['READ_minlen'] is not None:
-    read_min_len = config['READ_minlen']
-elif 'TRIMMOMATIC_minlen' in config and config['TRIMMOMATIC_minlen'] is not None:
-    read_min_len = config['TRIMMOMATIC_minlen']
-else:
-    print('ERROR in ', config_path, ': READ_minlen variable is empty. Please, complete ', config_path)
+read_min_len = validate_required_key(config, 'READ_minlen')
+if read_min_len < 50:
+    read_min_len = 50
 
-if type(read_min_len) != int:
-    print('ERROR in ', config_path, ': READ_minlen variable is not an integer. Please, complete ', config_path)
+# FIX
+#elif 'TRIMMOMATIC_minlen' in config and config['TRIMMOMATIC_minlen'] is not None:
+#    read_min_len = config['TRIMMOMATIC_minlen']
 
```

Comment:
`TRIMMOMATIC_minlen` seems to be deprecated, should it not be entirely removed?

Reply:
These three lines are removed later with the integration of config-parser module. Looks like I lied when I said "no commits cancelling each other".